cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -199,15 +199,9 @@ func (p *BicepProvider) Deploy(
 						timer.Stop()
 						return
 					case <-timer.C:
-						progressReport, err := progressDisplay.ReportProgress(ctx, &queryStartTime)
-						if err != nil {
-							// We don't want to fail the whole deployment if a progress reporting error occurs
-							log.Printf("error while reporting progress: %s", err.Error())
-							continue
+						if progressReport, err := progressDisplay.ReportProgress(ctx, &queryStartTime); err == nil {
```

Comment:
Can we keep the error logging if `err != nil`? I think that will still be helpful for debugging.

Reply:
The error is not very helpful.  At least the error we were looking before this PR. 
It was something like `"error while reporting progress: deployment not found"`.

Then, the error would keep repeating in the logs every ~10 seconds. (6 per min).
For a long `provision` like APIM that takes ~1hour, the logs would contain ~360 errors about `deployment not found`

At the end, it is more noise than help

Reply:
@vhvb1989 I think as a general design, we should log all silent errors where we don't know what happened.

In your particular case, it helped you link the progress report failing to deployment not found, then examined logs around "404" for a deployment query, and found the particular resource where querying failed.

In some other cases, it could be a client-side fault within `azd`. Not reporting it means we have no clue what actually triggered the error.

We should think about volume of logging separately. A count of 360 errors is a small number (easily greppable). In an extreme tight loop, we could think about introducing error logging deduplication -- but this is sophistication way beyond what's needed.

Reply:
Feel free to introduce a small map to avoid repeated logging of the same failure if it helps. The memory footprint is small enough in practice to be noticeable.

Reply:
added the log back